client/tooltool.py

```diff
@@ -720,25 +732,24 @@ def fetch_file(base_urls, file_record, grabchunk=1024 * 4, auth_file=None, regio
 
         # Well, the file doesn't exist locally.  Let's fetch it.
         try:
-            req = Request(url)
-            _authorize(req, auth_file)
-            f = urllib2.urlopen(req)
-            log.debug("opened %s for reading" % url)
-            with open(temp_path, mode="wb") as out:
-                k = True
-                size = 0
-                while k:
-                    # TODO: print statistics as file transfers happen both for info and to stop
-                    # buildbot timeouts
-                    indata = f.read(grabchunk)
-                    out.write(indata)
-                    size += len(indata)
-                    if len(indata) == 0:
-                        k = False
-                log.info("File %s fetched from %s as %s" %
-                         (file_record.filename, base_url, temp_path))
-                fetched_path = temp_path
-                break
+            with request(url, auth_file) as f:
+                with open(temp_path, **open_attrs) as out:
```

Comment:
why are you changing the way we open the file here ? if this patch is about calling @retry it seems like you are doing more here